src/nvim/charset.c

```diff
@@ -1918,10 +1932,9 @@ int hex2nr(int c)
 /// character, assume that all multi-byte characters are valid file name
 /// characters.
 ///
-/// @param str
-///
-/// @return true if `str` starts with a backslash that should be removed.
+/// @param  str  file path string to check
 bool rem_backslash(const char_u *str)
+  FUNC_ATTR_CONST FUNC_ATTR_WARN_UNUSED_RESULT FUNC_ATTR_NONNULL_ALL
```

Comment:
@bfredl PURE, but not because it depends on `chartab`. It dereferences `str`, so even if it did not depend on it or if `chartab` was `const` (I do not see any harm in CONST function examining `const` global memory: in any case, if somebody forgot the function code itself lies in global memory) (i.e. if BACKSLASH_IN_FILENAME is not defined) it would be PURE.